Tools/scripts/build_options.py

```diff
@@ -194,6 +194,8 @@ def __init__(self,
     Feature('Sensors', 'AIRSPEED', 'AP_AIRSPEED_ENABLED', 'Enable Airspeed Sensors', 1, None),    # Default to enabled to not annoy Plane users   # NOQA: E501
     Feature('Sensors', 'BEACON', 'BEACON_ENABLED', 'Enable Beacon', 0, None),
     Feature('Sensors', 'GPS_MOVING_BASELINE', 'GPS_MOVING_BASELINE', 'Enable GPS Moving Baseline', 0, None),
+    Feature('Sensors', 'HAL_WITH_DSP', 'HAL_WITH_DSP', 'Enable Library functions for Digital Signal Processing (including onboard-FFT analysis)', 0, None),  # noqa
```

Comment:
can we avoid exposing HAL_WITH_DSP to users? Its an implementation detail

Reply:
Done.  This also meant changing the hwdef files for several boards which were specifying HAL_WITH_DSP rather than HAL_GYROFFT_ENABLED